### PR TITLE
Feat/multi test docs

### PR DESCRIPTION
**Pull request by 0xFable**
Add documentation for `cw-multi-test` with specific sections for two popular implementations. 
Happy to make changes or add something that may be missing. Would like to try and write a really nice multi-test doc as I initially struggled with it and its really helpful for testing your contracts.

### CODE REVIEW

**Comment by 0xFable** [comment]
Interesting error on checks action, any ideas ? Maybe I missed a lint ?

**Comment by the-frey** [comment]
@0xFable looks like a link that doesn't resolve correctly?

**Comment by orkunkl** [comment]
IMO It would better to walk through multi test over a code example

**Comment by 0xFable** [comment]
@the-frey aha! That would make sense I tried to link to another page in this page. No worries will try to fix up.

**Comment by 0xFable** [comment]
@orkunkl will do, I have an idea for this to keep it simple

**Comment by 0xFable** [comment]
Added some more walkthrough like content. Still going through each concept one by one but now bringing all the concepts together with a test for cw-template. Have not tested that this all works yet will try do that soon. If it does should I add this int test to cw-template as well ?

**Comment by orkunkl** [comment]
Instead of writing tests for cw-template, I would recommend writing tests for a used simple contract. [cw20-pot](https://github.com/InterWasm/cw-contracts/tree/main/contracts/cw20-pot) makes most sense, since we use it in [dev academy](https://docs.cosmwasm.com/dev-academy/intro) so we can integrate tests it in dev-academy content pretty easily.

**Comment by 0xFable** [comment]
I could do this but this would remove 1 challenge from the tutorial ;) 
https://docs.cosmwasm.com/dev-academy/develop-smart-contract/develop#tests

image

**Comment by 0xFable** [comment]
Sounds like I am on the right path with the doc overall tho and I just need to tighten up the examples. Will keep whittling away until its nice

**Review by orkunkl — APPROVED** [review]

File: docs/04-smart-contracts/13-testing.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+All of that can be taken away almost in preference of cw-multi-test-based integration tests which enable you to test the flows and interactions between smart contracts. There is still a place for the flow described above but I have had a better experience writing these integration tests once you figure out the intricacies of multi-test. I hope to clear some of those intricacies up here with some tips, resources and steps.
+
+## `cw-multi-test` concepts
+
+There are a few main concepts to understand before you will be able to  simulate a blockchain environment in Rust and run tests that involve contract -> contract,
+and contract -> bank interactions.
+
+In this section we will take a step-by-step look through writing a test with cw-multi-test, explaining some important concepts along the way. 
+To start we need a specimen contract such as the [cw-template](https://github.com/InterWasm/cw-template/blob/main/src/contract.rs) which is a simple boilerplate contract containing two functions: `Increment` and `Reset`. While there are already unit tests for the cw-template, there is no `cw-multi-test` tests to simulate an entire run through.
```
Comment (orkunkl) [inline docs/04-smart-contracts/13-testing.md]: Update: integration tests has been added to the cw-template https://github.com/InterWasm/cw-template/pull/105
```suggestion
To start we need a specimen contract such as the [cw-template](https://github.com/InterWasm/cw-template/blob/main/src/contract.rs) which is a simple boilerplate contract containing two functions: `Increment` and `Reset`.
```

**Review by orkunkl — APPROVED** [review]

**Review by orkunkl — APPROVED** [review]

**Pull request merged**